keycloak-iam/src/main/java/com/valtimo/keycloak/service/KeycloakUserManagementService.java

```diff
@@ -23,24 +23,27 @@
 import com.ritense.valtimo.contract.authentication.model.ValtimoUser;
 import com.ritense.valtimo.contract.authentication.model.ValtimoUserBuilder;
 import com.ritense.valtimo.contract.utils.SecurityUtils;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import javax.ws.rs.NotFoundException;
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
-import javax.ws.rs.NotFoundException;
-import org.keycloak.representations.idm.RoleRepresentation;
-import org.keycloak.representations.idm.UserRepresentation;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.data.domain.Page;
-import org.springframework.data.domain.Pageable;
+
 import static com.ritense.valtimo.contract.Constants.SYSTEM_ACCOUNT;
 
 public class KeycloakUserManagementService implements UserManagementService {
     private static final Logger logger = LoggerFactory.getLogger(KeycloakUserManagementService.class);
+    protected static final int MAX_USERS = 1000;
```

Comment:
Isn't this just stretching the problem? @Klaas-Ritense already noticed the current situation was very slow. This will cause it to reach loadbalancer-time-limits.

Reply:
I've changed it. 
For the user list on the case page, it now no longer retrieves the roles for every user. 
For the user list in a user-task, it still retrieves the roles for every user, but it now does it for all users in parallel, by using `parallelStream()`